pymc_experimental/bart/bart.py

```diff
@@ -29,23 +29,20 @@ class BARTRV(RandomVariable):
 
     name = "BART"
     ndim_supp = 1
-    ndims_params = [2, 1, 0, 0, 0, 1]
+    ndims_params = [2, 1, 0, 0, 1]
     dtype = "floatX"
     _print_name = ("BART", "\\operatorname{BART}")
     all_trees = None
 
-    def _shape_from_params(self, dist_params, rep_param_idx=1, param_shapes=None):
-        return default_supp_shape_from_params(
-            self.ndim_supp, dist_params, rep_param_idx, param_shapes
-        )
-
-    def _infer_shape(cls, size, dist_params, param_shapes=None):
-        dist_shape = (cls.X.shape[0],)
-        return dist_shape
+    def _supp_shape_from_params(self, dist_params, rep_param_idx=1, param_shapes=None):
+        return (self.X.shape[0],)
 
     @classmethod
-    def rng_fn(cls, rng=np.random.default_rng(), *args, **kwargs):
-        return np.full_like(cls.Y, cls.Y.mean())
+    def rng_fn(cls, rng, X, Y, m, alpha, split_prior, size):
+        if size is not None:
+            return np.full((size[0], cls.Y.shape[0]), cls.Y.mean())
```

Comment:
Is size limited to one dimension on purpose? If so you could raise when BART variables are raised with size that ate longer than one, like `size=(2, 4)`

Reply:
Yes, that's on purpose. Good point!

Reply:
Usually size checks would be made in `make_node` or `dist`